src/main/groovy/com/twcable/gradle/cqpackage/CqPackageUtils.groovy

```diff
@@ -0,0 +1,216 @@
+        }
+
+        log.debug "OSGi project bundles: ${allProjectJars.sort()}"
+
+        return allProjectJars
+    }
+
+    /**
+     * Does the project use the "osgi" plugin?
+     */
+    static boolean isOsgiProject(Project project) {
+        if (project == null) throw new IllegalArgumentException("project == null")
+        return project.plugins.findPlugin('osgi') != null
+    }
+
+    /**
+     * Returns all jar files generated by the project dependencies in the configuration that use the "osgi" plugin.
+     */
+    @Nonnull
+    static Collection<File> projectJars(@Nonnull Configuration configuration) {
+        if (configuration == null) throw new IllegalArgumentException("configuration == null")
+
+        def projectDependencies = projectDependenciesForConfiguration(configuration)
+        def allProjectJars = jarsForProjectDependencies(projectDependencies)
+
+        log.debug "Project JARs: ${allProjectJars.sort()}"
+
+        return allProjectJars
+    }
+
+
+    @Nonnull
+    private static Collection<File> jarsForProjectDependencies(Collection<ProjectDependency> projectDependencies) {
+        return projectDependencies.collectMany { ProjectDependency dep ->
+            jarsForArtifacts(dep.projectConfiguration)
+        }
+    }
+
+
+    @Nonnull
+    static Collection<File> jarsForArtifacts(@Nonnull Configuration configuration) {
+        if (configuration == null) throw new IllegalArgumentException("configuration == null")
+        def allArtifacts = configuration.allArtifacts
+        def fileCollection = allArtifacts.files
+        def setOfFiles = fileCollection.files
+        setOfFiles.findAll { JAR_NAME_PATTERN.matcher(it.name).matches() }
+    }
+
+
+    @Nonnull
+    private static Collection<ProjectDependency> projectDependenciesForConfiguration(Configuration configuration) {
+        if (configuration == null) throw new IllegalArgumentException("configuration == null")
+
+        DependencySet dependencies = configuration.allDependencies
+
+        def projDeps = dependencies.findAll { it instanceof ProjectDependency } as Collection<ProjectDependency>
+
+        log.debug "Project Dependencies: ${projDeps.collect { it.dependencyProject.path }.sort()}"
+
+        return flattenProjectDependencies(projDeps)
+    }
+
+    /**
+     * Recurse through the project dependencies and return a set of every one encountered.
+     */
+    @Nonnull
+    private static Set<ProjectDependency> flattenProjectDependencies(Collection<ProjectDependency> projDeps) {
+        if (projDeps == null || projDeps.isEmpty()) return EMPTY_SET
+
+        // not very efficient, but works.
+        // assumes there's no cycles, which would be caught earlier by Gradle
+        Set<ProjectDependency> allProjDeps = new HashSet<>(projDeps)
+        projDeps.each { ProjectDependency dep ->
+            allProjDeps.addAll(projectDependenciesForProjDep(dep))
+        }
+        return allProjDeps
```

Comment:
That would be missing the seeded `projDeps`.

Though
```groovy
return projDeps.collect(new HashSet(projDeps)) { projectDependenciesForProjDep(it)} as Set<ProjectDependency>
```
would work. At that point, though, it's a bit much on one line and is no longer "conventional", therefore loosing its value compared to the `each`. Actually, I'd argue that `for (ProjectDependency dep : projDeps) {` would be better because it's more idiomatic. (Trivially more performant, too. But I'm more concerned about clarity.)